common/src/main/java/io/druid/math/expr/Evals.java

```diff
@@ -40,4 +46,39 @@ public static Number toNumber(Object value)
     }
     return longValue;
   }
+
+  public static boolean isConstant(Expr expr)
+  {
+    return expr instanceof DoubleExpr || expr instanceof LongExpr || expr instanceof StringExpr;
```

Comment:
Why is this not `instanceof ConstantExpr`? It seems like there is no point to having "ConstantExpr" as an interface if we can't use it to figure out if something is a constant or not.